android/src/main/java/com/novoda/notils/devicedetection/X86Detector.java

```diff
@@ -0,0 +1,45 @@
+package com.novoda.notils.devicedetection;
+
+import android.annotation.TargetApi;
+import android.os.Build;
+
+import java.util.Arrays;
+
+class X86Detector implements Detector {
+
+    private static final String X86 = "x86";
+
+    private final String[] supportedAbis;
+
+    public static Detector newInstance() {
+        AndroidVersion androidVersion = AndroidVersion.newInstance();
+        if (androidVersion.is21LollipopOrOver()) {
+            return new X86Detector(getSupportedAbis());
+        } else {
+            return new X86Detector(getLegacySupportedAbis());
+        }
+    }
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    private static String[] getSupportedAbis() {
+        return Build.SUPPORTED_ABIS;
+    }
+
+    private static String[] getLegacySupportedAbis() {
+        return new String[]{Build.CPU_ABI, Build.CPU_ABI2};
+    }
+
+    X86Detector(String... supportedAbis) {
+        this.supportedAbis = Arrays.copyOf(supportedAbis, supportedAbis.length);
```

Comment:
why make the copy here? Makes the class harder to test - should be done in the newInstance method so that the constr can be used easily for testing